File: app/src/main/java/com/example/android/newsapp/refactor/usecase/GetNewsUseCase.java
package com.example.android.newsapp.refactor.usecase;

import com.example.android.newsapp.refactor.repository.NewsGateway;

public class GetNewsUseCase implements UseCase {
    private final NewsGateway repository;
    private final DataCallback callback;

    public GetNewsUseCase(NewsGateway repository, DataCallback callback) {
        this.repository = repository;
        this.callback = callback;
    }

    @Override
    public void execute() {
        repository.getNews(callback);
    }
}

File: app/src/main/kotlin/com/example/android/newsapp/usecase/GetNewsUseCase.kt
package com.example.android.newsapp.usecase

import com.example.android.newsapp.model.Result
import com.example.android.newsapp.repository.NewsGateway
import com.example.android.newsapp.usecase.base.DataCallback
import com.example.android.newsapp.usecase.base.UseCase

class GetNewsUseCase(val repository: NewsGateway, val callback: DataCallback<List<Result>>) : UseCase {
    override fun execute() = repository.getNews(callback)
}
